### PR TITLE
[SQL][Doc] Fix documentation of lpad

### PR DESCRIPTION
**Pull request by actuaryzhang**
## What changes were proposed in this pull request?
Fix incomplete documentation for `lpad`.

### CODE REVIEW

**Review by srowen — APPROVED** [review]

**Comment by shaneknapp** [comment]
i will retrigger this once jenkins restarts

**Comment by shaneknapp** [comment]
test this please

**Comment by actuaryzhang** [comment]
jenkins, retest this please

**Comment by shaneknapp** [comment]
test this please

**Comment by shaneknapp** [comment]
test this please

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -2350,7 +2350,7 @@ object functions {
   def unbase64(e: Column): Column = withExpr { UnBase64(e.expr) }
 
   /**
-   * Right-padded with pad to a length of len.
+   * Right-pad the string column with pad to a length of len.
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: While here, let's make the description more complete in both cases, and consistent with the SQL usage doc:

```
    _FUNC_(str, len, pad) - Returns `str`, right-padded with `pad` to a length of `len`.
      If `str` is longer than `len`, the return value is shortened to `len` characters.
```

**Comment by actuaryzhang** [comment]
OK. Updated the doc as suggested. Thanks.

**Comment by srowen** [comment]
Merged to master/2.2

**Pull request closed without merging**